src/gluonnlp/models/bart.py

```diff
@@ -0,0 +1,376 @@
+            self.pooler = nn.Dense(units=self.units,
+                                   in_units=self.units,
+                                   flatten=False,
+                                   activation=pooler_activation,
+                                   weight_initializer=weight_initializer,
+                                   bias_initializer=bias_initializer)
+
+    def hybrid_forward(self, F, src_data, src_valid_length, tgt_data, tgt_valid_length):
+        """
+
+        Parameters
+        ----------
+        F
+        src_data
+            - layout = 'NT'
+                Shape (batch_size, src_length)
+            - layout = 'TN'
+                Shape (src_length, batch_size)
+        src_valid_length
+            Shape (batch_size,)
+        tgt_data
+            - layout = 'NT'
+                Shape (batch_size, tgt_length)
+            - layout = 'TN'
+                Shape (tgt_length, batch_size)
+        tgt_valid_length
+            Shape (batch_size,)
+
+        Returns
+        -------
+        (contextual_embedding)
+            - layout = 'NT'
+                Shape (batch_size, tgt_length, units)
+            - layout = 'TN'
+                Shape (tgt_length, batch_size, units)
+        (pooled_output)
+            This is optional. Shape (batch_size, units)
+        (dec_out)
+            - layout = 'NT'
+                Shape (batch_size, tgt_length, tgt_vocab_size)
+            - layout = 'TN'
+                Shape (tgt_length, batch_size, tgt_vocab_size)
+        """
+        enc_out = self.encode(F, src_data, src_valid_length)
+        contextual_embedding = self.decode_seq(F, tgt_data, tgt_valid_length, enc_out, src_valid_length)
+        if self.use_pooler:
+            pooled_output = self.apply_pooling(contextual_embedding)
+            return contextual_embedding, pooled_output
+        else:
+            dec_out = self.tgt_final_layer(contextual_embedding)
+            return dec_out
+
+    def apply_pooling(self, sequence):
+        """Generate the representation given the inputs.
+
+        This is used for pre-training or fine-tuning a mobile bert model.
+        Get the first token of the whole sequence which is [CLS]
+
+        sequence:
+            Shape (batch_size, sequence_length, units)
+        return:
+            Shape (batch_size, units)
+        """
+        outputs = sequence[:, 0, :]
+        if self.classifier_activation:
+            return self.pooler(outputs)
+        else:
+            return outputs
+
+    @property
+    def layout(self) -> str:
+        return self._layout
+
+    @property
+    def vocab_size(self):
+        return self._vocab_size
+
+    @classmethod
+    def get_cfg(cls, key=None):
+        if key is None:
+            return fair_bart_base()
+        else:
+            return bart_cfg_reg.create(key)
+
+    @classmethod
+    def from_cfg(cls, cfg,
+                 use_pooler=False,
+                 classifier_activation=False):
```

Comment:
Is it possible to remove the `classifier_activation` flag?

Reply:
We could remove this and consider applying an activation layer as default setting for the pooler.